### ISSUE TITLE
[feature] Add an ability to show outdated packages in conan install

### ISSUE THREAD

**Issue by mathbunnyru** [conan-io/conan#7778]
The idea is that it is quite a good idea to always use latest version (performance / security issues and so on)

For example I have libA, which I really need to use.
And I do install it using conan install.

Then new version gets released, but it's quite hard to follow manually all the packages updates.
I would like conan install to tell me that:
`You're using libA version 1.0, version 2.0 is available`.

It gets worse, when some library is used as a dependency and it's version is hardcoded in the conanfile.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by uilianries** [comment]
@mathbunnyru Do you mean [deprecated attribute](https://docs.conan.io/en/latest/reference/conanfile/attributes.html#deprecated), Or only related to package version?

For version we have a similar feature, but used for [conan config](https://docs.conan.io/en/latest/reference/config_files/conan.conf.html#conan-conf), the `config_install_interval`. In this case, we could run `conan search` every X time and check for any new versions. Or, create a new hook which checks it on pre-export.

**Comment by mathbunnyru** [comment]
No, I mean there's an openssl package, for example.

It had versions till 1.1.1g.
But recently 1.1.1h was added.

1.1.1g and below are not really deprecated in terms of conanfile.
But there's a newer version available.
So I want conan to tell me - 1.1.1h is available, please, update.

**Comment by jgsogo** [comment]
Hi, @mathbunnyru 

As uilianries suggested, it is probably something that cannot be built-in into the client and better fits in a hook (not sure if any of the current hooks is good fit for this functionality, but we can add more hook-functions if needed). Checking for new revisions/versions in all the remotes could take time and it is something we cannot run with every command (daily?) for all the packages in a graph (all packages in the cache?). There is also one additional problem, not all references use semver, the version is just an string and sometimes it is hard to know which one is newer.

Alternatively, it is something easy to implement in a CI, a Jenkins job can run daily (with custom functions to compare versions) and notify about new versions/revisions to the teams subscribed to these notifications. If you are using some server like Artifactory, you can store properties associated with every package that list the requirements, you can use these properties to know who you need to notify (the owner of a reference if some of its requirements have new versions/revisions).

IMHO, implementing this in a different job it is easier and more powerful than adding something to the client itself. wdyt?

**Comment by mathbunnyru** [comment]
> and sometimes it is hard to know which one is newer

I think there might be an assumption that versions are sorted in conandata file.

Implementing this feature on CI side looks fine to me, but it will require more actions from the user-side.

**Comment by jgsogo** [comment]
Regarding `conan-center-index` yes, we can rely on `conandata.yml` file and ordering (there can be bugs as ordering is manual). Nevertheless, that file is not uploaded to the remote as it is, only the lines corresponding to the version are uploaded with the recipe, run next line to check:

```
conan get openssl/1.1.1g@ conandata.yml
```

So, the server (or the client calling a server) knows only the references and cannot rely on any other info but the version itself to determine the ordering.

**Comment by mathbunnyru** [comment]
Oh, I see, thanks for the explanation.

**Comment by jgsogo** [comment]
IMO, from the client perspective, the best way to go would be to use a hook. In terms of usability it is almost the same to activate some configuration option or to install and enable a hook, and using a hook is more flexible if some library requires a custom function to compare recipe versions.

Using a CI and notifications... everything is user-side, nothing to add here.

Notifying from the server when a new version is available... stay tuned 😉

**Comment by mathbunnyru** [comment]
Another idea came to my mind.

When I upload a new version to CCI, the previous versions could be automatically marked deprecated (or something like that).
So, if someone uses them, they will get a warning output, if they have some flag specified.


I'm not sure at all if it's possible to do by the way with the current design :)

**Comment by memsharded** [comment]
Conan 2.2 has merged https://github.com/conan-io/conan/pull/15838 that implements the ``conan graph outdated`` command for this purpose.
